### DISCUSSION TITLE
Allow for multiple NC_DOMAIN domain names

### DISCUSSION THREAD

**Discussion by KenBW2** [nextcloud/all-in-one#2598] [Ideas] [answered]
I've always had my Nextcloud accessible on two domains - one a personal domain and one a work one. They point to the same NC instance, it's literally two Apache VirtualHosts for the same directory.

If I'm sharing a document, I want to be able to use the appropriate domain name

I can't see a way to do this in AIO. I imagine specifying multiple values for NC_DOMAIN will cause me a world of problems

### Describe the solution you'd like

I'd like to be able to either:
- specify an array of domains I can access NC AIO on
- have AIO not care what domain I'm accessing it on

### Describe alternatives you've considered

I run AIO behind a reverse proxy. I have both domains set up in the host's Apache config, but NC is redirecting the second domain to the first primary one and Collabora is having issues

### Additional context
None

**Comment by szaimen** [comment, marked as answer]
One AIO instance is indeed not meant to be used together with multiple domains. However you can follow https://github.com/nextcloud/all-in-one/blob/main/multiple-instances.md to create mutiple AIO instances, one for each domain.

**Reply by KenBW2** [reply]
> One AIO instance is indeed not meant to be used together with multiple domains. However you can follow https://github.com/nextcloud/all-in-one/blob/main/multiple-instances.md to create mutiple AIO instances, one for each domain.

Thanks for that. I did consider the multiple instances, but it seems overkill to duplicate the entire instance when NC supports it built in

FWIW I do use the manual-install/latest.yml

**Reply by szaimen** [reply]
> FWIW I do use the manual-install/latest.yml

You may simply remove this value then: https://github.com/nextcloud/all-in-one/blob/3fbd54989ab7a9dc5f42648142b57544a36e4ec4/manual-install/latest.yml#L77

But as stated above, better would be separate instances per domain.

**Reply by KenBW2** [reply]
> You may simply remove this value then:
> 
> https://github.com/nextcloud/all-in-one/blob/3fbd54989ab7a9dc5f42648142b57544a36e4ec4/manual-install/latest.yml#L77

Oh interesting. Remove the line entirely? Or just everything after the = character?

**Reply by david-alvarez-rosa** [reply]
Deleting the line altogether fixes it. Then add the second domain to the `trusted_domains` array in `config.php`.

**Comment by Fred-DTV** [comment]
I would like to have multiple domains with our AIO instance as well. 
May I ask how it went for you @KenBW2 ?

I am happy to try it and report any errors that come up too, but it would be nice to know what to expect what could go wrong

**Comment by burnclouds** [comment]
I want to use nextcloud.<domain> and nc.<domain> going to the same instance and having letencrypt generate the certificate for both. So i also would like to know how it went.

**Comment by wonder-wanderer** [comment]
@szaimen I know it I've read the docs, however I need to force mutiple domains in some instances even if I brake something.

So kindly please guide me here by giving some general-quick-and-dirty answers to my following two questions, so I know where to dig myself, this would help a lot, much appreciated!

**1) What would break in AIO if multiple domains are configured?**

Because for some reason multiple domains was removed from AIO design.  Would it break Talk, Collabora, Backup/Restore, Letsencrypt, Apache sites, ....? Regarding Letsencrypt I can just stick to reverse proxy mode.

**2) What are the methods to force multiple domains** and what would each method break?

My guesses here are:
a) Probably this should work? `occ config:system:set trusted_domains N --value=your.domain` where N would 3,4,5,6,.....
b) I see  ADDITIONAL_TRUSTED_DOMAIN in some places in the code, so it sets up "_occ config:system:set trusted_domains **2**_" and I hope also Collabora or Talk or anything that's needed
c) any other method?

**Comment by wonder-wanderer** [comment]
**3) OPTIONAL QUESTION: Is it ok to have multiple trusted domains in Nextcloud, and Collabora or Talk still operating on the main domain?** 

I mean if configuring multiple domains in Collabora or Talk is cumbersome, then would/should it work to configure the extra domains only in Nextcloud, but keep Collabora or Talk using just the main domain internally? Meaning that in web browser or Android app I login to Nextcloud via extradomain3.com, and that's what I see in web browser address bar even if I open a document in Collabora, but the connection to Collabora (in frontend browser javascript and/or backend Nextcloud server PHP) is still made via maindomain.com; and also if on Android I add the Talk app to the account logged in as extradomain3.com, would Talk internally use maindomain.com?

I stated this question as optional, because it requires knowledge not only abou configuring the integration between Nextcloud and Talk and Collabora, but perhaps knowing how then this connects internally inside the web page and Android app. But I hoped it might inspire some helpful thoughts anyway.
